syntax = "proto3";

import "Delta.proto";

// A message across a stream.
message ForwardMsg {
  oneof type {
    ////
    // Report Deltas
    ////

    NewConnection new_connection = 1;
    string new_report = 2;
    DeltaList delta_list = 3;
    string error = 4;
    bool report_finished = 5;

    ////
    // Upload Progresst
    ////

    // Indicates the % progres of a report upload.
    uint32 upload_report_progress = 6;

    // Indicates that the upload is complete and gives the URL.
    string report_uploaded = 7;
  }
}

// This is the first message that is sent when the connection is opened.
message NewConnection {
  // True if saving is properly configured.
  bool saving_configured = 1;
}
